session-manager/src/session.rs

```diff
@@ -43,27 +43,30 @@ pub struct Session {
 impl Session {
     /// Creates a new session from a server configuration and a list of
     /// principals.
-    pub fn new(config: rustls::ServerConfig, principals: &Vec<Principal>) -> Self {
-        let tls_session = ServerSession::new(&std::sync::Arc::new(config));
+    pub fn new(config: rustls::ServerConfig, principals: &Vec<Principal>) -> Result<Self, SessionManagerError> {
+        let mut tls_connection = ServerConnection::new(std::sync::Arc::new(config))?;
+        tls_connection.set_buffer_limit(Some(512 * 1024));
```

Comment:
Are these numbers arbitrary, or do they correspond to something?

Reply:
removed it as it's no longer necessary